test/initial_draft.cpp

```diff
@@ -153,12 +149,13 @@ class chain {
                     return std::move(_segment).invoke(receiver, STLAB_FWD(args)...);
                 };
             }
-        });
+        }, std::tuple_cat(std::move(_tail), std::tuple{std::move(_head)}));
     }
 
 public:
     template <class... Args>
-    using result_type = decltype(result_type_helper()(std::declval<Args>()...));
+    using result_type = 
```

Comment:
@sean-parent this is my best attempt at getting result_type to work, so far. If I try to use `result_type_helper` like this:

```cpp
    template <class... Args>
    using result_type = 
      decltype(std::declval<chain>().result_type_helper()(std::declval<Args>()...));
```

We fail by attempting to access a member of an incomplete type. Not sure if there's a cleaner way around this.

Reply:
I'm liking it! What more would you like to do on this before I push.

Reply:
If you're happy, I'm happy. Feel free to merge.